### PR TITLE
Showcase page

### LINKED ISSUE #287: Widget Showcase
**Enhancement**

We should have a widget showcase, where all widgets are available to view via a single page.  This should also be deployed to gh-pages for the repo, and therefore available on https://dojo.github.io/widgets/

### PR DESCRIPTION
**Pull request by bitpshr**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

This PR introduces a showcase page that leverages GitHub pages, the Dojo 2 CLI, and Travis by way of grunt-dojo2-extras. Efforts were focused on architecture instead of design for this iteration. This _should_ Just Work™ since I left the existing `deploy_key.enc` intact, but we may have to fiddle configuration settings for the widgets repo in Travis.

Example: bitpshr.github.io/widgets/

This work uncovered the following issues which should be addressed separately:

1. #295
2. #296
3. #297
4. #298
5. #299
6. #300
7. dojo/widget-core#661

Resolves #287

### CODE REVIEW

**Comment by bitpshr** [comment]
Friendly PR ping. Party on.

File: .gitignore [hunk 1]
```diff
@@ -12,5 +13,9 @@ coverage-final.lcov
 npm-debug.log
 yarn.lock
 /_apidoc
-deploy_key
 package-lock.json
+/.grunt/
+/.sync
+deploy_key
+deploy_key.pub
+.DS_Store
```
Comment (kitsonk) [inline .gitignore]: Traditionally we don't add this (`.DS_store`) to a repos `.gitignore` but instead manage these on a machine level (by creating or adding to `git config --global core.excludesfile`)

File: package.json [hunk 2]
```diff
@@ -28,6 +28,8 @@
     "@dojo/widget-core": "beta2"
   },
   "devDependencies": {
+    "@dojo/cli": "^2.0.0-beta2.3",
```
Comment (kitsonk) [inline package.json]: are these now `next` packages?

File: src/combobox/styles/comboBox.m.css.d.ts [hunk 3]
```diff
@@ -1,13 +1,13 @@
-export const clear: string;
-export const clearable: string;
+export const root: string;
 export const controls: string;
-export const disabledOption: string;
-export const dropdown: string;
 export const input: string;
-export const invalid: string;
-export const open: string;
+export const clearable: string;
```
Comment (kitsonk) [inline src/combobox/styles/comboBox.m.css.d.ts]: Why the churn on these files?  Was it related to a `tcm` run?

Comment (bitpshr) [reply src/combobox/styles/comboBox.m.css.d.ts]: _sigh_ my editor alphabetizes these automatically, I always forget to disable this feature. I'll blame it on `tcm` just for the heck of it though :)

**Comment by bitpshr** [comment]
Thanks @kitsonk. This has been rebased and should be ready for another review.

**Pull request merged**